presto-main/src/main/java/io/prestosql/execution/resourcegroups/InternalResourceGroup.java

```diff
@@ -669,6 +674,32 @@ private void startInBackground(ManagedQueryExecution query)
         }
     }
 
+    public void updateGroupsAndProcessQueuedQueries()
+    {
+        synchronized (root) {
+            updateResourceUsageAndGetDelta();
+
+            while (internalStartNext()) {
+                // start all the queries we can
+            }
+        }
+    }
+
+    public void generateCpuQuota(long elapsedSeconds)
+    {
+        synchronized (root) {
+            if (elapsedSeconds > 0) {
+                internalGenerateCpuQuota(elapsedSeconds);
+            }
+        }
+    }
+
+    // indirection for testing
+    protected void triggerProcessQueuedQueries()
```

Comment:
Is this only for testing? If so, consider making this public, marking @VisibleForTesting, and removing this comment.